### ISSUE TITLE
Sensitivity of the service

### ISSUE THREAD

**Issue by stefanreiterer** [OpenReqEU/prs-improving-requirements-quality#2]
The service is very "picky" regarding the wording of requirements, so it's difficult to write requirement descriptions that fulfill all the criteria that the service expects to be fulfilled. Our suggestion would be to have a parameter that allows the callee of the service to defining how sensitive the service should respond.
for example:
Parameter: sensitivity = 1
e.g. the service checks just simple criteria like text length 

....

Parameter: sensitivity = 10
e.g. the service checks all the criteria that are included at the moment

Best regards,
Stefan

**Comment by LloydThinks** [comment]
Hi Stefan,

Yes, the service currently has a big net of "catch all" that needs to be addressed. How to arbitrarily assign into categories such as "sensitivity" in itself would be difficult and probably lead to user confusion regarding what sensitivity means.

One idea we have is to allow configuration regarding which underlying algorithms should be triggered. This could easily be integrated as a configuration object that can be passed to filter out those algorithms. This is on my radar and will likely happen in some time.

However, the service is designed to show "suggestions" regarding quality, and it is up to the user how to interpret and utilise those suggestions. If you are implementing a user-interface on top of this component, you should add the ability for people to mark certain suggestions or words as "do not show me this again," and save that choice in a user object in your application. Because this API is stateless, I cannot save information like that (nor should it). Then, once you have this user-preferences object, you can filter out the suggestions the user does want yourself. Eventually, you may be able to pass that information to the API to filter out for you, but ultimately the result would be the same. So I recommend you put such an implementation in place yourself for your front-end implementation, and later, if such configurations are added to this API, then you can pass that along to the API to slightly reduce the running time of the algorithm.

TLDR: How to limit suggestions is arbitrary, so I recommend you offer user-based preferences.

Let me know if that makes sense.

Lloyd

**Comment by jnerlich** [comment]
@stefanreiterer is this a valid solution for you?

**Comment by stefanreiterer** [comment]
Hi Lloyd,

Of course user preferences can be implemented to filter the output of the service, but this includes the disadvantage of higher data transfer, a longer running time for the algorithms and higher implementation efforts for developers who want to use the service. Further I’m not sure if it’s good to let users determine how picky the service is, I see this more on a project or company level where the service can be used to create some sort of standard for the requirements quality. 

The idea with the configuration of the underlaying algorithms sounds very valuable to me :)

Best regards,
Stefan


 <http://www.selectionarts.com/>
 <x-msg://10/www.selectionarts.com>
DI Dr. Stefan Reiterer / Geschäftsführer
+43 (0)664 3123353 / s.reiterer@selectionarts.com <mailto:s.reiterer@selectionarts.com>
SelectionArts Intelligent Decision Technologies GmbH 
Radetzkystraße 10 / A-8010 Graz
www.selectionarts.com <x-msg://10/www.selectionarts.com>
This e-mail message may contain confidential or legally privileged information and is intended only for the use of the intended recipient(s). Any unauthorized disclosure, dissemination, distribution, copying or the taking of any action in reliance on the information herein is prohibited. E-mails are not secure and cannot be guaranteed to be error free as they can be intercepted, amended, or contain viruses. Anyone who communicates with us by e-mail is deemed to have accepted these risks. Company Name is not responsible for errors or omissions in this message and denies any responsibility for any damage arising from the use of e-mail. Any opinion and other statement contained in this message and any attachment are solely those of the author and do not necessarily represent those of the company.

**Comment by LloydThinks** [comment]
Hi Stefan,

Thank you for the feedback. I have marked down the configuration of the tool via algorithms as a "to do soon" and I'll also look into some kind of sensitivity assignment at the same time.

Perhaps a simple frequency analysis across a large, random set of requirements would allow me to report and gauge the sensitivity numbers. However, I fear this is not the right direction as the most common errors are perhaps the ones that need to be addressed the most, even if there are a lot of them. Some more thinking to be done for sure.

Lloyd

**Comment by LloydThinks** [comment]
@stefanreiterer,

I have update the code to include a config option that you pass in the JSON to to the API. Please see the updated [Readme](https://github.com/OpenReqEU/prs-improving-requirements-quality/blob/master/README.md) for instructions on how to do this.

I am still considering the original request of this issue (sensitivity of results), and will keep you updated.

Lloyd

**Comment by jnerlich** [comment]
@stefanreiterer did you have the chance to have a look at @LloydThinks  changes?

**Comment by stefanreiterer** [comment]
Hi Jennifer,

sorry for my late answer  - I was on the road for the last 3 days. Michael and me took a look at the changes before the Hamburg meeting and we think that they absolutely make sense and improve the usability requirements quality component. Unfortunately we had not enough time lift adapt the integration that we already had in Censio according to the changes.

Best regards
Stefan 
 <http://www.selectionarts.com/>
 <x-msg://219/www.selectionarts.com>
DI Dr. Stefan Reiterer / Geschäftsführer
+43 (0)664 3123353 / s.reiterer@selectionarts.com <mailto:s.reiterer@selectionarts.com>
SelectionArts Intelligent Decision Technologies GmbH 
Radetzkystraße 10 / A-8010 Graz
www.selectionarts.com <x-msg://219/www.selectionarts.com>
This e-mail message may contain confidential or legally privileged information and is intended only for the use of the intended recipient(s). Any unauthorized disclosure, dissemination, distribution, copying or the taking of any action in reliance on the information herein is prohibited. E-mails are not secure and cannot be guaranteed to be error free as they can be intercepted, amended, or contain viruses. Anyone who communicates with us by e-mail is deemed to have accepted these risks. Company Name is not responsible for errors or omissions in this message and denies any responsibility for any damage arising from the use of e-mail. Any opinion and other statement contained in this message and any attachment are solely those of the author and do not necessarily represent those of the company.

**Comment by jnerlich** [comment]
Thanks for having a look at it. I will then close this issue. 
Please feel free to reopen if you find any issues with it.
